src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -119,7 +119,7 @@ private T runWithRetries(final int slot, int attempts, boolean tryRandomNode, Je
         this.connectionHandler.renewSlotCache();
       }
 
-      return runWithRetries(slot, attempts - 1, tryRandomNode, redirect);
+      return runWithRetries(slot, attempts - 1, true, null);
```

Comment:
This would heavily affect current applications. Connection exceptions can happen for silliest of the reasons. So we can't backoff right after a connection exception. There should a balance between retry and backoff. For example, a slot based node or a redirected node should be given at least 2/3 consecutive tries before backing off.

In all of the cases, `maxAttempts`, remaining `attempts`, `renewSlotCache()` should be kept in mind.

Reply:
I'll just try to rephrase what you said so I know I understand:
* With this change, we'd start trying random nodes on the first connection exception
* In your opinion, what we should do is:
  1. First, try the exact same operation a few times in a row
  1. If that doesn't help, start trying random nodes

I think this sounds reasonable, just want to ensure I change the right thing!

Reply:
@walles Yes, you've got it right.